### PR TITLE
Try a cron based approach

### PR DESCRIPTION
**Pull request by lakrisgubben**

### CODE REVIEW

**Comment by redundans** [comment]
Great improvments! I did add a cancel function as fall back if adding cron fails. To test it. Start export and run `wp cron event delete eep_create_export`. This will trigger a state where you have to use the cancel link to be able to generate export again.

**Comment by lakrisgubben** [comment]
> Great improvments! I did add a cancel function as fall back if adding cron fails. To test it. Start export and run `wp cron event delete eep_create_export`. This will trigger a state where you have to use the cancel link to be able to generate export again.

💯

**Comment by lakrisgubben** [comment]
@redundans added a filter so different products can control which meta should be added to the total content. Here is what I'd use for LO

```PHP
add_filter(
	'eep_meta_included_in_content',
	function(): array {
		return [ 'preamble' ];
	}
);
```

**Comment by lakrisgubben** [comment]
@redundans also added this to remove all content in noscript tags https://github.com/klandestino/export-editionprotocol/pull/1/commits/31583612f50dc833ca8fcc5a29072a913dad26a9 can't think of any scenario where we'd want that content to be counted since its not visible on the site. :)

**Comment by redundans** [comment]
@lakrisgubben Pushed a new concept for the filter (renamed) `eep_included_in_content`.

File: classes/Formatter.php [hunk 1]
```diff
@@ -56,21 +55,30 @@ private function get_issue_year( WP_Post $post ): string {
...
 	 */
 	private function get_post_content( WP_Post $post ): string {
-		$post_excerpt      = get_post_field( 'post_excerpt', $post );
-		$post_content      = get_post_field( 'post_content', $post );
-		$meta_content      = '';
-		$included_meta     = apply_filters( 'eep_meta_included_in_content', [] );
-		if ( ! empty( $included_meta ) ) {
-			foreach( $included_meta as $key ) {
-				$meta_content .= get_post_meta( $post->ID, $key, true );
+		$included_in_content = apply_filters( 'eep_included_in_content', [ 'post_field' => 'post_content' ] );
```
Comment (lakrisgubben) [inline classes/Formatter.php]: Maybe I'm reading this wrong, but shouldn't it be an array of arrays, i.e. `'post_field' => [ 'post_content', ... ]` to be able to set more than one post_field, meta or taxonomy?

And then something like

```PHP
foreach ( $included_in_content as $type => $keys ) {
    foreach ( $keys as $key ) {
        switch ( $type ) {
       ....
```

Comment (redundans) [reply classes/Formatter.php]: You are absolutly right. Did not cross my mind. 🥸

**Pull request merged**